Tests/MSFT_xPackageResource.TestHelper.psm1

```diff
@@ -68,148 +66,396 @@ function New-MockFileServer
+                [String]
+                $LogFile,
 
-        # Wait for client to finish downloading
-        $pipe = New-Object -TypeName 'System.IO.Pipes.NamedPipeServerStream' -ArgumentList @( '\\.\pipe\dsctest2' )
-        $pipe.WaitForConnection()
-        $pipe.Dispose()
+                [Parameter(Mandatory = $true)]
+                [String]
+                $Message
+            )
+            
+            $Message >> $LogFile
+        }
 
-        $response.Dispose()
-        $httpListener.Stop()
-        $httpListener.Close()
+        # End of function declarations - Beginning of function execution
 
-        # Close pipe
-    
-        # Use net shell command to clean up the certificate binding
-        netsh http delete sslcert ipport=0.0.0.0:1243
-    }
+        if ($null -eq (Get-NetFirewallRule -DisplayName 'UnitTestRule' -ErrorAction 'SilentlyContinue'))
+        {
+            New-NetFirewallRule -DisplayName 'UnitTestRule' -Direction 'Inbound' -Program "$PSHome\powershell.exe" -Authentication 'NotRequired' -Action 'Allow'
+        }
 
-    netsh advfirewall set allprofiles state on
-}
+        $null = netsh advfirewall set allprofiles state off
 
-<#
-    .SYNOPSIS
-        Creates a new test executable.
+        Write-Log -LogFile $LogPath -Message (Get-Date)
 
-    .PARAMETER DestinationPath
-        The path at which to create the test executable.
-#>
-function New-TestExecutable
-{
-    [CmdletBinding()]
-    param
-    (
-        [Parameter(Mandatory = $true)]
-        [ValidateNotNullOrEmpty()]
-        [String]$DestinationPath
-    )
-    
-    if (Test-Path -Path $DestinationPath)
-    {
-        Write-Verbose -Message "Removing old executable at $DestinationPath..."
-        Remove-Item -Path $DestinationPath -Force
-    }
+        $HttpListener = New-Object 'System.Net.HttpListener'
+        $fileServerStarted = $null
 
-    $testExecutableCode = @'
-    using System;
-    using System.Collections.Generic;
-    using System.Linq;
-    using System.Management;
-    using System.Text;
-    using System.Threading.Tasks;
-    using System.Management.Automation;
-    using System.Management.Automation.Runspaces;
-    using System.Runtime.InteropServices;
-    namespace Providers.Package.UnitTests.MySuite
-    {
-        class ExeTestClass
+        try
         {
-            public static void Main(string[] args)
+            # Set up the listener
+            if ($Https)
             {
-                string cmdline = System.Environment.CommandLine;
-                Console.WriteLine("Cmdline was " + cmdline);
-                int endIndex = cmdline.IndexOf("\"", 1);
-                string self = cmdline.Substring(0, endIndex);
-                string other = cmdline.Substring(self.Length + 1);
-                string msiexecpath = System.IO.Path.Combine(System.Environment.SystemDirectory, "msiexec.exe");
+                $HttpListener.Prefixes.Add([Uri]'https://localhost:1243')
```

Comment:
I would add a comment in the documentation of Start-Server saying it listens on https://localhost:1243 & http://localhost:1242.

Clients need to know this in order to use the mock server